### PR TITLE
CAP-172 connect edr server to object store

### PR DESCRIPTION
**Pull request by crbunney**
Some fixes to support the integration of this code with the edr_server

### CODE REVIEW

File: src/clean_air/models.py [hunk 1]
```diff
@@ -37,7 +37,15 @@ class Metadata:
...
     def id(self):
-        return self.dataset_name.lower().replace(string.punctuation, "_")
+        id_str = self.name
+        for c in string.punctuation.replace("_", ""):
+            if c in id_str:
+                id_str = id_str.replace(c, "")
+        for c in string.whitespace:
+            if c in id_str:
+                id_str = id_str.replace(c, "_")
+        return id_str.lower()
```
Comment (corinnebosley) [inline src/clean_air/models.py:47]: I don't really understand what this bit does but I'm assuming it's required and tested, and therefore looks fine to me.

File: tests/unit/test_models.py [hunk 2]
```diff
@@ -7,20 +9,34 @@
...
+            if c in id_str:
+                id_str = id_str.replace(c, "")
+        for c in string.whitespace:
+            if c in id_str:
+                id_str = id_str.replace(c, "_")
+        return id_str.lower()
+
     def test_init_defaults(self):
-        test_dataset_name = "test_dataset"
+        test_dataset_name = "MOASA Flight M270"
```
Comment (corinnebosley) [inline tests/unit/test_models.py:26]: Is this the correct name for the flight? I thought they were all MOSSA flights but I could be wrong.

**Review by corinnebosley — APPROVED** [review]
Most of it looks pretty good to me, if anything at all needs changing then it is the flight name (MOASA/MOSSA?) but other than that minor detail I approve.

File: src/clean_air/models.py (hunk 1 above)
Comment (crbunney) [reply src/clean_air/models.py:47]: basically stripping out special characters and whitespace from the string we use as an ID so the result is more URL friendly.

(I'd prefer to use something like an integer for the ID, but without a central database issuing them, it's extra complexity to ensure they're unique. I could use a [GUID](http://guid.one/guid), but they're not terribly user friendly)

File: tests/unit/test_models.py (hunk 2 above)
Comment (crbunney) [reply tests/unit/test_models.py:26]: I don't know, but I took it from the name of the data files, so if it is wrong, maybe I've misunderstood something?

File: src/clean_air/models.py (hunk 1 above)
Comment (corinnebosley) [reply src/clean_air/models.py:47]: Right, I see.  Well it seems fine for now at least.  Maybe we can think about integers or GUIDs at a later date when we have more systems in place, but doesn't seem necessary right now.  Good job!

File: tests/unit/test_models.py (hunk 2 above)
Comment (corinnebosley) [reply tests/unit/test_models.py:26]: I guess you're probably right then.  It's not even a one-line change if it needs fixing anyway, and we'll find out if it does.

**Pull request merged**